src/imageResource.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+require('openseadragon');
+
+var ImageResource = function(config) {
+  this.needed = config.needed || false;
+  this.visible = config.visible || false;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.height = config.height || 1;
+  this.width = config.width || 1;
+  this.zIndex = config.zIndex || 0;
+  this.tileSource = config.tileSource;
+  this.dynamic = config.dynamic || false;
+  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', or 'detail'
+  this.status = 'initialized'; // can be 'requested', 'received', 'pending','shown', or 'failed'
+  this.parent = config.parent;
+  this.dispatcher = config.dispatcher;
+};
+
+ImageResource.prototype = {
+  hide: function() {
+    this.previousOpacity = this.opacity;
```

Comment:
My comment still stands here: 

> Probably a better way to handle this would be to have an `_updateOpacity` method that keys off of both the `opacity` and `visible` properties. It would be called from hide, show, and setOpacity. That way you don't have to mess with previousOpacity and worry about it getting stale. 

As it is, you won't get the right result in this scenario: 

```
resource.hide();
resource.setOpacity(0.5);
resource.show();
```

At least in my mind, what should happen is that the resource should remain hidden when you call setOpacity, but when you show it'll be 0.5. The way the code is now, on the setOpacity line it will become visible, but its `visible` will still be false.